language-plutus-core/src/Language/PlutusCore/Pretty/PrettyConst.hs

```diff
@@ -0,0 +1,39 @@
+{-# LANGUAGE DefaultSignatures    #-}
+{-# LANGUAGE FlexibleInstances    #-}
+{-# LANGUAGE OverloadedStrings    #-}
+{-# LANGUAGE TypeSynonymInstances #-}
+
+module Language.PlutusCore.Pretty.PrettyConst where
+
+import qualified Data.ByteString.Lazy               as BSL
+import           Data.Foldable                      (fold)
+import qualified Data.Text                          as T
+import           Data.Text.Prettyprint.Doc
+import           Data.Text.Prettyprint.Doc.Internal (Doc (Text))
+import           Data.Word                          (Word8)
+import           Numeric                            (showHex)
+
+class PrettyConst a where
+    prettyConst :: a -> Doc ann
+    default prettyConst :: Show a => a -> Doc ann
+    prettyConst = pretty . show
```

Comment:
Are you sure you want to have a default implementation in terms of `Show` rather than `Pretty`?

Reply:
I'm not sure.  That's definitely what we want for strings and characters, but those may be the only types for which the difference matters.  The `parseConst` method isn't used yet, but we'll presumably want `parseConst . prettyConst` to be the identity (or something close), and I'm not sure if there's a suitable default for `parseConst` if the default implementation for `prettyConst is `pretty`.  Maybe we shouldn't have defaults and then people will be forced to think about it.